servicetalk-http-api/src/main/java/io/servicetalk/http/api/RedirectConfigBuilder.java

```diff
@@ -432,10 +432,18 @@ public StreamingHttpRequest apply(final boolean relative,
         private static void fullCopy(final StreamingHttpRequest originalRequest,
                                      final StreamingHttpRequest redirectRequest) {
             redirectRequest.setHeaders(originalRequest.headers());
-            redirectRequest.transformMessageBody(p -> p.ignoreElements().concat(originalRequest.messageBody()));
-            // Use `transform` to update PayloadInfo flags, assuming trailers may be included in the message body
-            redirectRequest.transform(NOOP_TRAILERS_TRANSFORMER);
-            // FIXME: instead of `transform`, preserve original PayloadInfo/FlushStrategy when it's API is available
+            preserveMessageBody(originalRequest, redirectRequest);
+            // Sync PayloadInfo
+            if (DefaultStreamingHttpRequest.class.equals(redirectRequest.getClass()) &&
```

Comment:
Since DefaultStreamingHttpRequest is final, would instance of also work here? (I always wonder if instanceof is quicker or this class check)

Reply:
It will, but I'm not sure if it has any optimizations for `final` classes. I only remember that for non-final classes and interfaces it's slower. Even if it has, I believe it will optimize it to `class.equals`.